examples/tree_1d_dgsem/elixir_advection_PERK2.jl

```diff
@@ -0,0 +1,65 @@
+
+using Convex
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the linear advection equation
+
+advection_velocity = 1.0
+equations = LinearScalarAdvectionEquation1D(advection_velocity)
+
+# Create DG solver with polynomial degree = 3 and (local) Lax-Friedrichs/Rusanov flux as surface flux
+solver = DGSEM(polydeg = 3, surface_flux = flux_lax_friedrichs)
+
+coordinates_min = -1.0 # minimum coordinate
+coordinates_max = 1.0 # maximum coordinate
+
+# Create a uniformly refined mesh with periodic boundaries
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 4,
+                n_cells_max = 30_000) # set maximum capacity of tree data structure
+
+# A semidiscretization collects data structures and functions for the spatial discretization
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition_convergence_test,
+                                    solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+# Create ODE problem with time span from 0.0 to 1.0
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan);
+
+# At the beginning of the main loop, the SummaryCallback prints a summary of the simulation setup
+# and resets the timers
+summary_callback = SummaryCallback()
+
+# The AnalysisCallback allows to analyse the solution in regular intervals and prints the results
+analysis_callback = AnalysisCallback(semi, interval = 100)
+
+# The StepsizeCallback handles the re-calculation of the maximum Δt after each time step
+stepsize_callback = StepsizeCallback(cfl = 2.5)
+
+alive_callback = AliveCallback(alive_interval = 1)
```

Comment:
This seems small, how many steps do we take?

Reply:
Currently 13, but I can change this to 2.7335 which is the value of CFL calculated from dtmax. This results in 12 timesteps.

Reply:
Ah sorry, this is a misunderstanding - I meant the `alive_interval`, not the `cfl` number.

Hm, in that case we should increase the endtime to take at least a couple hundred timesteps to be really sure that we are stable.

Reply:
I have changed tspan to (0.0, 20.0) so that there are 256 timesteps. Moreover I tried also to change `dtmax`  in `method_PERK2.jl` (for timespan in optimization to 1.0 or 2.0, however, this seems to yield minimal effect on the result from `analysis_callback(sol)`